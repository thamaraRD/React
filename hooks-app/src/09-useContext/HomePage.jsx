import { useContext } from "react";
import { UserContext } from "./context/UserContext";



export const HomePage = () => {

  const { user } = useContext( UserContext );
  
    return (
      <div>
        <h1>HomePage de { user?.nombre }</h1>
        <hr />
        <pre aria-label="pre">
          {JSON.stringify(user, null, 3)}
        </pre>
      </div>
    );
  };