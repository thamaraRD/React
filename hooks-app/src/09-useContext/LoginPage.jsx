import { useContext } from "react";
import { UserContext } from "./context/UserContext";

export const LoginPage = () => {

  const { user, setUser } = useContext( UserContext );

    return (
      <div>
        <h1>LoginPage</h1>
        <hr />

      <pre aria-label="pre">
        { JSON.stringify( user, null, 2 ) }
      </pre>
      <button
      className="btn btn-success"
      onClick={() => setUser({ id: 222, nombre: 'Jesus', email: 'jesus@jesus.com' })}
      >
        Establecer usuario
      </button>

      </div>
    );
  };