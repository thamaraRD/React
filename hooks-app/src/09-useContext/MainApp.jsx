import { Navigate, Route, Routes } from "react-router-dom";
import { UserProvider } from "./context/UserProvider";
import { AboutPage } from "./AboutPage";
import { HomePage } from "./HomePage";
import { LoginPage } from "./LoginPage";
import { Navbar } from "./Navbar";

export const MainApp = () => {
  return (
    <UserProvider>
      <Navbar />
      <hr />

      <Routes>
        <Route path="/" element={ <HomePage /> } />
        <Route path="about" element={ <AboutPage /> } />
        <Route path="login" element={ <LoginPage /> } />

      {/* Ruta Comod√≠n */}
        <Route path="/*" element={ <Navigate to="/about" /> } />
      </Routes>
    </UserProvider>
  );
};
