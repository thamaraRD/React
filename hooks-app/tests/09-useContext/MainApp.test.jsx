import { render, screen } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom";
import { MainApp } from "../../src/09-useContext/MainApp"
import { Navbar } from "../../src/09-useContext/Navbar";

describe('testing en el componente <MainApp />', () => { 

  test('should de mostrar el HomePage', () => { 

    render(
  <MemoryRouter>
    <MainApp/>
  </MemoryRouter>
    )

    expect(screen.getByText('HomePage de')).toBeTruthy();
    

 });
  test('should de mostrar el LoginPage', () => { 

    render(
  <MemoryRouter initialEntries={['/login']}>
    <MainApp/>
  </MemoryRouter>
    )

    expect(screen.getByText('LoginPage')).toBeTruthy();


 });

 test('should match to snapshot navBar', () => { 

  const { container } = render(
  <MemoryRouter>
    <Navbar />
  </MemoryRouter>
  );
  expect(container).toMatchSnapshot();

  });

})