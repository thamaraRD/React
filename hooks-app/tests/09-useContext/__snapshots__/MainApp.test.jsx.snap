// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing en el componente <MainApp /> should match to snapshot navBar 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-dark rounded-3"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        UseContext
      </a>
      <div
        class="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          class="navbar-nav"
        >
          <a
            aria-current="page"
            class="nav-link active"
            href="/"
          >
            Home
          </a>
          <a
            class="nav-link "
            href="/about"
          >
            About
          </a>
          <a
            class="nav-link "
            href="/login"
          >
            Login
          </a>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
